segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/filesystem/FileSystemStorage.java

```diff
@@ -0,0 +1,377 @@
+            return supplyAsync(targetHandle.getSegmentName(),
+                    () -> syncConcat(targetHandle, offset, sourceSegment));
+    }
+
+    @Override
+    public CompletableFuture<Void> delete(SegmentHandle handle, Duration timeout) {
+            return supplyAsync(handle.getSegmentName(), () -> syncDelete(handle));
+    }
+
+    //endregion
+
+    //region AutoClosable
+
+    @Override
+    public void close() {
+
+    }
+
+    //endregion
+
+    //region private sync implementation
+
+    @SneakyThrows(StreamSegmentNotExistsException.class)
+    private SegmentHandle syncOpenRead(String streamSegmentName) {
+        Path path = Paths.get(config.getFilesystemRoot(), streamSegmentName);
+
+        if (!Files.exists(path)) {
+            throw new StreamSegmentNotExistsException(streamSegmentName);
+        }
+
+        return FileSystemSegmentHandle.getReadHandle(streamSegmentName);
+    }
+
+    @SneakyThrows
+    private SegmentHandle syncOpenWrite(String streamSegmentName) {
+        Path path = Paths.get(config.getFilesystemRoot(), streamSegmentName);
+        if (!Files.exists(path)) {
+            throw new StreamSegmentNotExistsException(streamSegmentName);
+        } else if (Files.isWritable(path)) {
+            return FileSystemSegmentHandle.getWriteHandle(streamSegmentName);
+        } else {
+                return openRead(streamSegmentName).get();
+        }
+    }
+
+    @SneakyThrows(IOException.class)
+    private int syncRead(SegmentHandle handle, long offset, byte[] buffer, int bufferOffset, int length) {
+
+        Path path = Paths.get(config.getFilesystemRoot(), handle.getSegmentName());
+
+            if (Files.size(path) < offset) {
+                log.warn("Read called on segment {} at offset {}. The offset is beyond the current size of the file.",
+                        handle.getSegmentName(), offset);
+                throw new IllegalArgumentException( "Reading beyond the current size of segment");
+            }
+
+        try (FileChannel channel = FileChannel.open(path, StandardOpenOption.READ)) {
+            int bytesRead = channel.read(ByteBuffer.wrap(buffer, bufferOffset, length), offset);
+            return bytesRead;
+        }
+    }
+
+    @SneakyThrows(IOException.class)
+    private SegmentProperties syncGetStreamSegmentInfo(String streamSegmentName) {
+            PosixFileAttributes attrs = Files.readAttributes(Paths.get(config.getFilesystemRoot(), streamSegmentName),
+                    PosixFileAttributes.class);
+            StreamSegmentInformation information = new StreamSegmentInformation(streamSegmentName, attrs.size(),
+                    !(attrs.permissions().contains(OWNER_WRITE)), false,
+                    new ImmutableDate(attrs.creationTime().toMillis()));
+
+            return information;
+    }
+
+    private boolean syncExists(String streamSegmentName) {
+        boolean exists = Files.exists(Paths.get(config.getFilesystemRoot(), streamSegmentName));
```

Comment:
Just return the result, don't store it in a variable. There are a few more instances of this in the file as well.